src/cpp/utils/threading/threading_win32.ipp

```diff
@@ -23,71 +24,87 @@ namespace eprosima {
 
 template<typename ... Args>
 static void set_name_to_current_thread_impl(
+        std::array<char, 16>& thread_name_buffer,
         const char* fmt,
         Args... args)
 {
-    char thread_name[16]{};
-    snprintf(thread_name, 16, fmt, args ...);
+    snprintf(thread_name_buffer.data(), 16, fmt, args ...);
 
     std::wstringstream stream;
-    stream << thread_name;
+    stream << thread_name_buffer.data();
     std::wstring w_thread_name = stream.str();
 
     SetThreadDescription(GetCurrentThread(), w_thread_name.c_str());
 }
 
 void set_name_to_current_thread(
+        std::array<char, 16>& thread_name_buffer,
         const char* name)
 {
-    set_name_to_current_thread_impl("%s", name);
+    set_name_to_current_thread_impl(thread_name_buffer, "%s", name);
 }
 
 void set_name_to_current_thread(
+        std::array<char, 16>& thread_name_buffer,
         const char* fmt,
         uint32_t arg)
 {
-    set_name_to_current_thread_impl(fmt, arg);
+    set_name_to_current_thread_impl(thread_name_buffer, fmt, arg);
 }
 
 void set_name_to_current_thread(
+        const char* fmt,
+        uint32_t arg)
+{
+    std::array<char, 16> thread_name_buffer;
+    set_name_to_current_thread(thread_name_buffer, fmt, arg);
+}
+
+void set_name_to_current_thread(
+        std::array<char, 16>& thread_name_buffer,
         const char* fmt,
         uint32_t arg1,
         uint32_t arg2)
 {
-    set_name_to_current_thread_impl(fmt, arg1, arg2);
+    set_name_to_current_thread_impl(thread_name_buffer, fmt, arg1, arg2);
 }
 
 static void configure_current_thread_priority(
+        const char* thread_name,
         int32_t priority)
 {
     if (priority != std::numeric_limits<int32_t>::min())
     {
         if (0 == SetThreadPriority(GetCurrentThread(), priority))
         {
             EPROSIMA_LOG_ERROR(SYSTEM,
-                    "Error '" << GetLastError() << "' configuring priority for thread " << GetCurrentThread());
+                    "Problem to set priority of thread with id [" << GetCurrentThreadId() << "," << thread_name << "] to value " << priority <<
```

Comment:
Please keep the calls to `GetLastError()` in log messages for Win32 implementation